application/src/main/java/org/opentripplanner/transit/service/TimetableRepositoryIndex.java

```diff
@@ -233,6 +258,25 @@ private void initalizeServiceCodesForDate(TimetableRepository timetableRepositor
     }
   }
 
+  private void initializeTheEndOfServiceDateForStop() {
+    for (StopLocation stop : patternsForStop.keySet()) {
```

Comment:
I know that here isn't a precedent in this class, but can you generate an intermediate HashMap and call `Map.copyOf` at the end of this method. That will create an immutable Map which is very space-efficient.